packages/ui-kit/src/components/Counter/utils.ts

```diff
@@ -6,13 +8,16 @@ interface getValueOptions {
 const getValue = (options: getValueOptions) => {
   const { value, localize } = options
 
-  if (value !== null && Number.isInteger(parseFloat(value))) {
-    return localize ? parseInt(value).toLocaleString() : parseInt(value)
-  }
   if (value === null) {
     return '-'
   }
 
+  if (getIsNonNumeric(value)) return value
+
+  if (value !== null && Number.isInteger(parseFloat(value))) {
+    return localize ? parseInt(value).toLocaleString() : parseInt(value)
+  }
+
   return localize
     ? parseFloat(parseFloat(value).toFixed(2).toLocaleString()).toLocaleString()
     : parseFloat(value).toFixed(2)
```

Comment:
I'm not sure what the behavior here should be. What are we trying to do?

1. Check if it's null?
2. Check if it's an integral number?
3. Check if it's a non-integral number?
4. Check if it's anything else?

Why not rewrite this:

```ts
// 1. Render `null` as "-".
if (value == null) {
  return '-'
}

const number = parseFloat(value)

// 2. Handle integral numbers.
if (Number.isInteger(number)) {
  return localize
    ? parseInt(value).toLocaleString()
    : parseInt(value)
}

// 3. Handle non-integral numbers.
if (Number.isFinite(number) && !Number.isNaN(number)) {
  return localize
    ? parseFloat(number.toFixed(2)).toLocaleString()
    : number.toFixed(2)
}

// 4. Handle anything else.
return value
```

Notice that we don't need `getIsNonNumeric` with this approach. Actually, we may not even want `getIsNonNumeric`, because we actually need to consider each of

- integral (`Number.isInteger`)
- non-integral
- NaN (`Number.isNaN`)
- +Infinity/-Infinity (`Number.isFinite`)

and `getIsNonNumeric` doesn't tell us that.

Reply:
Additionally, this looks like a good function to export and have unit tests for. I see you added high-level, component-level tests in Counter.test.tsx for this; but, IMO, it also makes sense to just test this function directly, for each type of input it might handle.

Maybe rename it, too? `getValueForCounter`?

Reply:
Yeah, that makes total sense to me, probably this function would also fit for the leaderboard and everywhere we need to use numeric values

Reply:
I would also expose a placeholder for `null` state as a component's prop to make it configurable, e.g. something like
```
onNullValue: (value) => {
  // Some custom logic here to handle null in an app
  return '-'
}
```

Reply:
I agree @sashathor 👍 However, we should perhaps make that change in a separate PR, considering also NaN and +Infinity/-Infinity. I filed https://linear.app/propel/issue/PRO-2572 for this.